DataValitation-MVPTests/RegistrationPresenterTests.swift

```diff
@@ -0,0 +1,259 @@
+//
+//  RegistrationPresenterTests.swift
+//  DataValitation-MVPTests
+//
+//  Created by Tatiana Magdalena on 03/02/18.
+//  Copyright © 2018 Tatiana Magdalena. All rights reserved.
+//
+
+import XCTest
+@testable import DataValitation_MVP
+
+class RegistrationPresenterTests: XCTestCase {
+    
+    var presenterUnderTest: RegistrationPresenter!
+    var mockView: MockValidationView!
+    
+    override func setUp() {
+        super.setUp()
+        
+        mockView = MockValidationView()
+        presenterUnderTest = RegistrationPresenter(view: mockView,
+                                                   nameValidator: NameValidator(),
+                                                   emailValidator: EmailValidator(),
+                                                   cpfValidator: CPFValidator())
+        mockView.presenter = presenterUnderTest
+    }
+    
+    override func tearDown() {
+        mockView = nil
+        presenterUnderTest = nil
+        super.tearDown()
+    }
+    
+}
+
+// MARK: - Name tests -
+
+extension RegistrationPresenterTests {
+    
+    func test_InvalidNameChange_CallHandlers() {
+        // given
+        let name = "Ab C"
+        // when
+        presenterUnderTest.userNameChanged(newName: name)
+        //then
+        XCTAssertEqual(mockView.nameHandlerCallCount, 1)
+        XCTAssertEqual(mockView.emailHandlerCallCount, 0)
```

Comment:
Ao escrever testes, é comum reparar que muitas vezes as asserções podem acabar sendo semanticamente as mesmas entre testes diferentes, com pequenas variações de parâmetros na prática.

Não seria interessante extrair essas quatro asserções em um método privado, com alguns named parameters designando bem as expectativas em relação à asserções?

Reply:
Vê se é isso que entendi:
A ideia seria remover essas asserções daqui e criar um método com elas, por exemplo:
```
private func assertHandlersCount(name: Int, email: Int, cpf: Int, readyToValidate: Int) { 
   XCTAssertEqual(...)
   ...
}
```
E aí nessa parte do teste eu chamo essa função. É isso?

Reply:
Mas se for, testei fazer isso agora, mas me deparei com um problema: 
Estou chamando esta função em várias partes do teste; se mais de um deles dá erro em algum desses Assert, eu fico sem saber de onde veio cada um. O Xcode só me diz que teve "n" falhas em cada assert.

Se não for isso que vc tava dizendo eu to viajando então...